server/testdriver/core/src/main/java/org/infinispan/server/test/api/RespTestClientDriver.java

```diff
@@ -37,8 +43,12 @@ public RedisClient get() {
       RedisClient client = RedisClient.create(resources, configuration.redisURI);
       client.setOptions(configuration.clientOptions);
       testClient.registerResource(() -> {
-         resources.shutdown(0, 15, TimeUnit.SECONDS).getNow();
-         client.shutdown(0, 15, TimeUnit.SECONDS);
+         try {
+            resources.shutdown(0, 15, TimeUnit.SECONDS).getNow();
+            client.shutdownAsync(0, 15, TimeUnit.SECONDS).get(15, TimeUnit.SECONDS);
+         } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            log.error("Timed out waiting RESP client to shutdown", e);
```

Comment:
This seems fine to avoid blocking indefinitely, but any idea what the shutdown is blocking in the first place?

Reply:
Agreed, it should never have blocked indefinitely before. It seems like a bug in lettuce and/or netty from what limited amount I looked at it.

Reply:
But this should be fine to integrate until/if we find the root cause.

Reply:
I went through the thread dump but found nothing concrete. So I guess that Lettuce was assuming that closing resources would be immediate, so either there is a bug in there while closing or in Netty blocking somewhere.

The threads that got my attention are the `pool-1-thread-1` and `2`. The `1` is blocking on `EPollPort` and the `2` is runnable on `EPoll.wait`. But I could not trace from where they spawn.